src/main/java/bio/terra/cli/auth/AuthenticationManager.java

```diff
@@ -187,22 +188,22 @@ public void populateCurrentTerraUser() {
   public void fetchPetSaCredentials(TerraUser terraUser) {
     // if the current workspace is not defined, then we don't know which pet SA to fetch
     if (workspaceContext.isEmpty()) {
-      logger.info("There is no current workspace defined.");
+      logger.debug("There is no current workspace defined.");
       return;
     }
 
     // if the key file for this user + workspace already exists, then no need to re-fetch
     Path jsonKeyPath = GlobalContext.getPetSaKeyFile(terraUser, workspaceContext);
-    logger.info("Looking for pet SA key file at: {}", jsonKeyPath);
+    logger.debug("Looking for pet SA key file at: {}", jsonKeyPath);
     if (jsonKeyPath.toFile().exists()) {
-      logger.info("Pet SA key file for this user and workspace already exists.");
+      logger.debug("Pet SA key file for this user and workspace already exists.");
     } else {
       // ask SAM for the project-specific pet SA key
       HttpUtils.HttpResponse petSaKeySamResponse =
           new SamService(globalContext.server, terraUser).getPetSaKeyForProject(workspaceContext);
       if (!HttpStatusCodes.isSuccess(petSaKeySamResponse.statusCode)) {
-        logger.info("SAM response to pet SA key request: {})", petSaKeySamResponse.responseBody);
-        throw new RuntimeException(
+        logger.debug("SAM response to pet SA key request: {})", petSaKeySamResponse.responseBody);
+        throw new InternalErrorException(
             "Error fetching pet SA key from SAM (status code = "
```

Comment:
[follow-up] Not for this PR, but as we look towards an external-facing release, it will make sense to scrub through these exceptions for acronyms or words that won't make sense to end users. For internal system errors, where we don't think they're user caused or user actionable, we may want to err on the side of more opaqueness in the user-facing message. (Of course there's a limit... I wouldn't just give a made-up error code, but possibly a generic message like "An error occurred while fetching Terra service credentials for your user account" combined with a more precise error locator code "(code 1234)".

Reply:
Good idea. Added this to the list of style improvements in [PF-574](https://broadworkbench.atlassian.net/browse/PF-574).